src/components/ZipCodeInput.js

```diff
@@ -1,42 +1,40 @@
-import { useState } from "react"
 import { Button, Stack, TextField } from "@mui/material"
 import SearchIcon from "@mui/icons-material/Search"
 import fetchPetList from "../components/fetchPetList"
 import { useContext } from "react"
 import { PetsContext } from "../context"
+import { SearchContext } from "../context"
 
 const ZipCodeInput = () => {
-  //using PetsContext
+  //using PetsContext and SearchContext
   const { setPetList } = useContext(PetsContext)
+  const { search, handleSearch } = useContext(SearchContext)
 
-  const [zipcodeFormValues, setZipcodeFormValues] = useState("")
-
-  //Gets value from the input and updates the zipCode state
+  //Validation. Gets value from the input and updates the zipCode state
   const handleZipcodeChange = (e) => {
     const userZipCode = e.target.value
     if (userZipCode.length <= 5) {
-      setZipcodeFormValues(userZipCode)
+      handleSearch(e)
     }
   }
-
-  //OnSubmit Form consoles the user input
+  //  makes a fetch request with current zip and selected animal type
+  //should handleSubmit be included in the SearchContext and moved to <App/>?
   const handleSubmit = (e) => {
     e.preventDefault()
-    fetchPetList(zipcodeFormValues).then((response) => {
+    fetchPetList(search.zipcode, search.animalType).then((response) => {
       setPetList(response.animals)
     })
-    setZipcodeFormValues("")
   }
 
```

Comment:
Should the `handleSubmit` function be moved to `<App/>` and added to `SearchContext`? Or should it just remain in `<ZipCodeInput/>`

Reply:
I think the handleSubmit function can remains in the on the <ZipCodeInput/> and move it up later, once we need it in another component.

Reply:
i'm working on the sorting and I need the zipcode to make the api calls. I was planning on making the zipcode as context, but now seeing you made SearchContext, will I be able to get this down to my sorting component if we move SearchContext to App? I will also now need animal type.

Reply:
`SearchContext.provider` is in `<App/>` and the values passed to it can be used in any child component since I wrapped it around all components in `<App/>`. 

The state passed to `SearchContext.provider` that  you're going to be interested in accessing is ` const [search, setSearch] = React.useState({ zipcode: "", animalType: "cat" })
`

To access this value in the component where you want to use , you will have to call the useContext() hook.